server/handlers/get_totp.go

```diff
@@ -0,0 +1,35 @@
+package handlers
+
+import (
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/keratin/authn-server/app"
+	"github.com/keratin/authn-server/app/services"
+	"github.com/keratin/authn-server/lib/route"
+	"github.com/keratin/authn-server/server/sessions"
+)
+
+//GetTOTP begins the TOTP onboarding process
+func GetTOTP(app *app.App) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// check for valid session with live token
+		accountID := sessions.GetAccountID(r)
+		if accountID == 0 {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		img, err := services.TOTPCreator(app.AccountStore, app.TOTPCache, accountID, route.MatchedDomain(r))
+		if err != nil {
+			panic(err)
+		}
+
+		w.Header().Set("Content-Type", "image/png")
+		w.Header().Set("Content-Length", strconv.Itoa(len(img.Bytes())))
```

Comment:
I would like to revisit this response so that it can be JSON containing a PNG. In the [previous discussion](https://github.com/keratin/authn-server/pull/133#discussion_r334215025) I imagined returning the URI as well for implementations that don't want to depend entirely on QR code scanning. Whether or not we return the URI or refactor this endpoint to create a JWT, a JSON response would give us more flexibility to revisit response requirements at a later date.